backend/coreapp/views/scratch.py

```diff
@@ -293,15 +272,25 @@ def compile(request, slug):
     scratch: Scratch = Scratch.objects.get(slug=slug)
 
     # Get the context from the backend if it's not provided
-    if not context:
+    if context == None:
         logger.debug("No context provided, getting from backend")
         context = scratch.context
 
-    result = CompilerWrapper.compile_code(compiler, compiler_flags, code, context)
+    try:
+        result = CompilerWrapper.compile_code(compiler, compiler_flags, code, context)
+    except CompilationError as e:
+        return APIException(str(e), code=str(status.HTTP_400_BAD_REQUEST))
 
-    diff_output: Optional[Dict[str, Any]] = None
-    if result.elf_object:
+    try:
         diff_output = AsmDifferWrapper.diff(scratch.target_assembly, scratch.platform, scratch.diff_label, result.elf_object)
+    except AssemblyError as e:
```

Comment:
unless you want to do something different for them, I think you can do 
```py
except (AssemblyError, DiffError, ObjdumpError, CompilationError) as e:
```

Reply:
yeah I wasn't sure yet if we should handle these differently. For now I'll do that, though